File: artifacts/tulipindicators/translate/src/crossany.rs
use crate::*;

pub fn ti_crossany_start(options: Option<&[f64]>) -> i32 {
    let _ = options; // Explicitly ignore the parameter to match C behavior
    1
}
pub fn ti_crossany(size: i32, inputs: &[&[f64]], options: &[f64], outputs: &mut [&mut [f64]]) -> i32 {
    let a = inputs[0];
    let b = inputs[1];
    let output = &mut outputs[0];
    let mut output_idx = 0;

    for i in 1..size as usize {
        let cross = ((a[i] > b[i]) && (a[i - 1] <= b[i - 1])) || ((a[i] < b[i]) && (a[i - 1] >= b[i - 1]));
        output[output_idx] = if cross { 1.0 } else { 0.0 };
        output_idx += 1;
    }

    0
}

File: artifacts/tulipindicators/decompose/src/crossany.c
int ti_crossany_start(const double *options)
{
  (void) options;
  return 1;
}


/*** DEPENDENCIES:
***/


int ti_crossany(int size, const double * const *inputs, const double *options, double * const *outputs)
{
  const double *a = inputs[0];
  const double *b = inputs[1];
  (void) options;
  double *output = outputs[0];
  unsigned int output_idx = 0;
  int i;
  for (i = 1; i < size; i += 1)
  {
    output[output_idx] = ((a[i] > b[i]) && (a[i - 1] <= b[i - 1])) || ((a[i] < b[i]) && (a[i - 1] >= b[i - 1]));
    output_idx += 1;
  }

  return 0;
}


/*** DEPENDENCIES:
***/


